File: app/src/main/java/net/squanchy/navigation/RoutingInjector.java
package net.squanchy.navigation;

import android.support.v7.app.AppCompatActivity;

import net.squanchy.injection.ActivityContextModule;
import net.squanchy.injection.ApplicationInjector;
import net.squanchy.navigation.deeplink.DeepLinkModule;
import net.squanchy.signin.SignInModule;
import net.squanchy.support.debug.DebugPreferencesModule;

final class RoutingInjector {

    private RoutingInjector() {
        // no instances
    }

    public static RoutingComponent obtain(AppCompatActivity activity) {
        return DaggerRoutingComponent.builder()
                .activityContextModule(new ActivityContextModule(activity))
                .applicationComponent(ApplicationInjector.obtain(activity))
                .deepLinkModule(new DeepLinkModule())
                .navigationModule(new NavigationModule())
                .signInModule(new SignInModule())
                .routingModule(new RoutingModule())
                .build();
    }
}

File: app/src/main/java/net/squanchy/navigation/RoutingInjector.kt
package net.squanchy.navigation

import android.support.v7.app.AppCompatActivity

import net.squanchy.injection.ActivityContextModule
import net.squanchy.injection.ApplicationInjector
import net.squanchy.navigation.deeplink.DeepLinkModule
import net.squanchy.signin.SignInModule

internal object RoutingInjector {

    fun obtain(activity: AppCompatActivity): RoutingComponent {
        return DaggerRoutingComponent.builder()
                .activityContextModule(ActivityContextModule(activity))
                .applicationComponent(ApplicationInjector.obtain(activity))
                .deepLinkModule(DeepLinkModule())
                .navigationModule(NavigationModule())
                .signInModule(SignInModule())
                .routingModule(RoutingModule())
                .build()
    }
}
